.wiki/Configuration-Storage.md

```diff
@@ -73,6 +74,43 @@ Artipie tests, check the [implementation](https://github.com/artipie/asto/blob/m
 for more details. There is no possibility to use in memory storage from configuration, 
 it's for unit and integration tests only.
 
+## Custom storage
+
+Artipie users have an option to implement and use an own storage.
+If you want to make your storage, you need to define `asto-code` dependency in `pom` file of your project:
+```xml
+<dependency>
+   <groupId>com.artipie</groupId>
+   <artifactId>asto-core</artifactId>
+   <version>...</version>
+</dependency>
+```
+On the next step, you have to implement interface 
+[Storage](https://github.com/artipie/asto/blob/master/asto-core/src/main/java/com/artipie/asto/Storage.java)
+to host data in the place which you need. The interface 
+[StorageFactory](https://github.com/artipie/asto/blob/master/asto-core/src/main/java/com/artipie/asto/factory/StorageFactory.java)
+is responsible for creating a new storage instance. You have to implement this interface and 
+mark the implementation by annotation [ArtipieStorageFactory](https://github.com/artipie/asto/blob/master/asto-core/src/main/java/com/artipie/asto/factory/ArtipieStorageFactory.java). 
+This annotation helps Artipie to find factory classes and provides a name of storage type. 
+Storage type name must be unique in the scope of one Artipie server. In the case of type name conflict, 
+Artipie will throw an exception on a start-up stage. Storage configuration is represented by an 
+interface [StorageConfig](https://github.com/artipie/asto/blob/master/asto-core/src/main/java/com/artipie/asto/factory/StorageConfig.java). 
+Currently, this interface has the single implementation 
+[YamlStorageConfig](https://github.com/artipie/asto/blob/master/asto-core/src/main/java/com/artipie/asto/factory/StorageConfig.java#L96) 
+that allows to define configuration as `yaml` file. It's also possible to use an own implementation of `StorageConfig`.
+
+To start Artipie with a custom storage, you have to:
+- provide a file with configuration of storage; this is done the same way as for other storage types.
+- put a jar file contained implementation classes and all needed libraries to the class path.
+
+If logging is switched to `info` level, you should see the following log record:
+```
+Initiated storage factory [type={your-storage-type}, class={your-storage-factory-class-name}]
+```
+
+You can study [a storage implementation based on Redisson](https://github.com/artipie/asto/tree/master/asto-redis/src/main/java/com/artipie/asto/redis)
```

Comment:
Are you sure, that Redis is called Redisson? As far as I know, Redisson are hotels...

Reply:
[Redisson](https://redisson.org/) is a `redis` client that is used to implement storage.